hog.py

```diff
@@ -248,8 +251,17 @@ def bacon_strategy(score, opponent_score):
     >>> bacon_strategy(50, 70)
     0
     """
-    "*** YOUR CODE HERE ***"
-    return 5 # Replace this statement
+
+    if (opponent_score < 7) or (opponent_score == 0):
+        return BASELINE_NUM_ROLLS
+    elif (opponent_score > 7) and (opponent_score < 10):
+        return 0
+    elif opponent_score > 10: 
```

Comment:
Rather than check the opponent's score 3 separate times, why not get the value of what _would_ be the score attained from using the free bacon rule based on the opponent's score and compare it with the BACON_MARGIN. Only if using the free bacon rule would result in the lower score than the BACON_MARGIN, would you return the BASELINE_NUM_ROLLS rather than 0. This way there are only 2 return statements in this function, one for each of the possible outcomes, rather than 4 return statements.

Reply:
Check the `take_turn` implementation so you can stick to the EAFP principle while calculating the value of free bacon.